### ISSUE TITLE
Could not find textures

### ISSUE THREAD

**Issue by sackboylbp** [overviewer/Minecraft-Overviewer#1485]
I'm not sure what I am doing wrong, I downloaded the latest version of Minecraft overviewer today and want to create a render of the world from my minecraft server. I downloaded the map onto my computer but when I run "overviewer.py /home/isaac/Downloads/VentureSurvival/world /home/isaac/" I get the message:

2018-10-01 21:44:39  Welcome to Minecraft Overviewer!
2018-10-01 21:44:40  Generating textures...
2018-10-01 21:44:40 E Could not find the textures while searching for 'assets/minecraft/textures/blocks/grass_top.png'. Try specifying the 'texturepath' option in your config file.
Set it to the path to a Minecraft Resource pack.
Alternately, install the Minecraft client (which includes textures)
Also see <http://docs.overviewer.org/en/latest/running/#installing-the-textures>
(Remember, this version of Overviewer requires a 1.12-compatible resource pack)
(Also note that I won't automatically use snapshots; you'll have to use the texturepath option to use a snapshot jar)

I have the minecraft client installed and 1.12.2.jar located at .minecraft/version/1.12.2. The documentation says it should automatically find it but that doesn't seem to be the case right now. Am I doing something wrong? Please help, thanks!

**Comment by netmonger** [comment]
I too run a server on a Linux box.  My quick fix for this was to create a directory under the home directory of the user that runs the server and where all the server files are stored.  The directory I created was just like the one a client would create.

.minecraft/versions/1.13.1/

Into that directory I moved the Minecraft client jar file, not the server jar file.  I downloaded the client jar file from minecraft.net and it usually comes in as the version number so in this case it was 1.13.1.jar.  Once that was in place overviewer.py started rendering.

**Comment by sackboylbp** [comment]
I did exactly as you said and get the same error. Is there any other way I can try to give it textures because it isn't getting them from my 1.12.2.jar. I will try it on another machine and get back to you.

**Comment by SkyDiverCool** [comment]
I am having the same issue on a personal windows device I've tried installing both 1.12 and 1.12.2 vanilla but it is still unable to find it

**Comment by netmonger** [comment]
@sackboylbp You can specify your own textures.  Look through this documentation http://docs.overviewer.org/en/latest/running/#installing-the-textures.

@SkyDiverCool Can you tell me the exact path to the 1.12.jar or 1.12.2.jar files on your Windows installs?

**Comment by IamSAB** [comment]
Just investigated the v1.13.1 jar file. The texture folder is not a direct child of assets. There is only a subfolder called minecraft where you can find textures and the other folders.

**Comment by IamSAB** [comment]
I copied .minecraft to dirs below, still not finding textures
- /home
- /opt/mscs
- /home/minecraft

**Comment by netmonger** [comment]
I am wondering if this is a version problem.  Are you guys using overviewer that was built for 1.13 or 1.12?  In either case the texture pack or client JAR file needs to be of the same version.

**Comment by IamSAB** [comment]
The docs do not mention you need different overviewer version for different minecraft versions?!
Of course I am using the latest overviewer installed like described here
http://docs.overviewer.org/en/latest/installing/#debian-ubuntu

**Comment by IamSAB** [comment]
Then this issue is related to the 1.13 version of minecraft in my case
https://github.com/overviewer/Minecraft-Overviewer/issues/1454

Probably you could point that out in the docs.

**Comment by Bobcat00** [comment]
Not sure if this will help you, but I'm using Overviewer 0.12.278 with Minecraft 1.12.2.  I tried to use the PureBDcraft texture pack, but Overviewer complained about not being able to find certain textures.

I solved it by copying the texture pack to a convenient location (same folder as the Overviewer config file), then opening the texture pack with WinRAR and opening the MC 1.12.2 client .jar with another instance of WinRAR.  I then found the textures folder in each jar file, selected the subfolders in the MC client jar, dragged (copied) them to the texturepack jar and told WinRAR to skip existing files.  This updated the texture pack jar with the additional/missing files from the client jar.

Not sure if this is your problem, but maybe it will help.

**Comment by digandbuilder** [comment]
Hello I also have this problem I have tried the solutions however the same issue is still given, has anyone figured this out yet and if so please tell me.

**Comment by shov** [comment]
Well, I've copied the client into /home/<user>/.minecraft/versions... and run overviewer on world dir, everything went smoothly
```
user@server:~/mcmap$ mkdir -p ~/.minecraft/versions/1.13.2
user@server:~/mcmap$ cp ../Uploads/1.13.2.jar ~/.minecraft/versions/1.13.2/
user@server:~/mcmap$ cd ..
user@server:~$ overviewer.py /home/user/spigot-1.13.2/world /home/user/mcmap/
2019-03-14 20:38:59  Welcome to Minecraft Overviewer!
2019-03-14 20:38:59  Generating textures...
2019-03-14 20:39:12  Preprocessing...
2019-03-14 20:39:25  Rendering 1241 total tiles.
```

**Comment by CounterPillow** [comment]
Well this issue has become a catch-all for everyone who either doesn't have the proper minecraft version installed or messed up something else, going "me too pls help" which is not something that allows anyone to actually help you.

In short: yes, you do need the right version of the client jar for overviewer to work, and no, we're not going to support older client jars for older worlds because you can render older worlds with newer client jars too.

**Comment by Bobcat00** [comment]
Ok, that’s fine. You might want to make it clearer on the web site that a particular release will only work with a particular MC version. Maybe even list which Overviewer version to use for older MC versions.

**Comment by CounterPillow** [comment]
I think "(Remember, this version of Overviewer requires a 1.14-compatible resource pack)" in the big red error message that people get that leads them towards this issue should be enough information to figure out which version your particular overviewer version is compatible with.

**Comment by XcomReborn** [comment]
Using Ubuntu 18.04 bionic

I have tried using the 1.14.jar client but this appears not to work. Using texturepath = "/home/user/texturepacks/texturepack.zip" appears to get through some of the textures but it always get to one of them that is missing and then throws the following error:

2019-07-07 21:53:32 E Could not find the textures while searching for 'assets/minecraft/textures/entity/shulker/shulker_light                                         _gray.png'. Try specifying the 'texturepath' option in your config file.
Set it to the path to a Minecraft Resource pack.
Alternately, install the Minecraft client (which includes textures)
Also see <http://docs.overviewer.org/en/latest/running/#installing-the-textures>
(Remember, this version of Overviewer requires a 1.14-compatible resource pack)
(Also note that I won't automatically use snapshots; you'll have to use the texturepath option to use a snapshot jar)

I've tried about 6 different texture packs and I can't get them to finish generating textures before it stops.

ideas ?
